students/Dkuchan/ClassProject/PythonClassProject.py

```diff
@@ -0,0 +1,625 @@
+        self.win = 1
+        self.gears = None
+        self.alliance = None
+        self.fieldpos = None
+        self.pilot = None
+        self.autogears = None
+        self.kpa = None
+        self.autokpa = None
+        self.climb = None
+        self.penalties = None
+
+    def __repr__(self):
+        return '{}, {}: {}'.format(self.team, self.matchnumber, self.win)
+
+    def __str__(self):
+        return 'Match {}, Team {}'.format(self.matchnumber, self.team)
+
+    #def __iter__(self):
+
+class MonteResult(Match):
+    # this class is intended to contain results for monte carlo runs
+    # it is a sublcass of Match as a monte run is just simulated Matches
+    # functions of match are not all used at this point.  perhaps with future updates
+
+    def __init__(self, bluescore, redscore, runnum):
+        self.run = None
+        self.bluescore = None
+        self.redscore = None
+        self.winner = None
+
+    def __str__(self):
+        return "Match {}, Winner {}, BlueScore: {}, RedScore: {}".format(self.run, self.winner, self.bluescore, self.redscore)
+
+
+class Team:    # Team is an object which contains data specific to a team
+    # it is not intended to be user editable but is user viewable
+
+    # number - Done
+    # name - Done
+    # matches played - Done
+    # matches on red - Done in alliances list (0 for blue, 1 for red)
+    # matches on blue - Done in alliances list (0 for blue, 1 for red)
+    # matches in pos 1  - Done in fieldpos (1,2,3) left to right
+    # matches in pos 2 - Done in fieldpos (1,2,3) left to right
+    # matches in pos 3 - Done in fieldpos (1,2,3) left to right
+    # most gears in a match - Done
+    # least gears in a match - Done
+    # average gears in a match - Done
+    # most KPA in a match - Done
+    # least KPA in a match - Done
+    # average KPA in a match - Done
+    # gears delivered in auto - Done (as dict) lets see if that was a good idea
+    # gears delivered in auto pos 1
+    # gears delivered in auto pos 2
+    # gears delivered in auto pos 3
+    # KPA in auto
+    # number of climbs
+    # number of pilots
+
+    def __init__(self, num):
+        self.number = num
+        self.name = None
+        self.matchesplayed = []
+        self.gearsdelivered = []
+        self.kpa = []
+        self.pilot = []
+        self.alliances = []
+        self.fieldpos = []
+        self.autogears = []
+        self.autogearsinpos = []
+        self.autokpa = []
+        self.autokpainpos = []
+        self.climb = []
+        self.matchwins = []
+
+   # def __repr__(self):
+    #    return '{}, {}: {}'.format(self.team, self.matchnumber, self.win)
+
+    def __str__(self):
+        return 'Team Number: {}, Team Name: {}'.format(self.number, self.name)
+
+    def fillmatches(self):
+        for i in range(0, len(matchlist)):   # I wanted to do this in a list
+            if int(matchlist[i].team) == int(self.number):  # did your team play in this match
```

Comment:
shouldn't these already be ints?

and you are only using i to pull out an item form the matchlist -- so you could just loop through the matchlist.

I _think_ this listcomp will do:

self.matchesplayed = [ match.number for match in matchlist if match.team == self.number ]